src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -961,11 +961,15 @@ private async Task<WorkUnitResult> ExecuteInstantiatedTask(ITaskExecutionHost ta
                     && !taskResult // and it returned false
                     && !taskLoggingContext.HasLoggedErrors // and it didn't log any errors
                     && (be is TaskHost th ? th.BuildRequestsSucceeded : false)
-                    && (be is IBuildEngine7 be7 ? !be7.AllowFailureWithoutError : true) // and it's not allowed to fail unless it logs an error
                     && !(_cancellationToken.CanBeCanceled && _cancellationToken.IsCancellationRequested)) // and it wasn't cancelled
                 {
+                    // If is allowed to fail without error
+                    if (be is IBuildEngine7 be7 && be7.AllowFailureWithoutError)
+                    {
+                        taskLoggingContext.LogComment(MessageImportance.Normal, "TaskReturnedFalseButDidNotLogError", _taskNode.Name);
```

Comment:
Fit-and-finish question (@baronfel, especially): This logs with the `MSB4181:` prefix. Generally, we try not to have "error codes" for message-level logging. But here, it's kind of a warning, if you squint really hard. Should we strip the prefix, or leave it as-is?

Reply:
The normal case is hopefully non-error (they rerouted the error(s) appropriately, so it's just an informational message they don't care about). Most of our other messages are similar to this in that they _might_ indicate an error but usually don't, so I'd vote for stripping the prefix, personally, but it's reasonable to leave it to baronfel.

Reply:
> Most of our other messages are similar to this in that they _might_ indicate an error but usually don't

I don't think I agree with this. We have a lot of messages that are fully neutral like "Started building target", and that feels like a very solid majority of logging in a normal build to me.

Reply:
I think of your example as "might indicate an error but usually don't"; if that target should not be building (or should not be building yet), it's indicating an error...but it's normally innocuous.